server/src/controllers/game.controller.js

```diff
@@ -107,8 +107,46 @@ class GameController {
   // }
 
   // End game logic by checkmate or draw
-  // async endGame (req, res){
-  // }
+  async endGame(req, res, next) {
+    try {
+      const { game_id, user_id } = req.body;
+      const isGameExist = await GameRepo.fetchOne(game_id);
+
+      if (!isGameExist.data) {
+        return res
+          .status(400)
+          .send(response("Game does not exist", null, false));
+      }
+
+      if (!user_id) {
+        return res
+          .status(400)
+          .send(response("User does not exist", null, false));
+      }
+
+      let permission;
+
+      const updated = await GameRepo.delete(game_id, {
```

Comment:
boss, this should be ``GameRepo.update`` not ``delete``

Reply:
Yeah... you should be updating the game status.